### PR TITLE
Audit/error handling

### PR DESCRIPTION
**Pull request by vysahjk**

### CODE REVIEW

File: src/controllers/category.controller.js [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
+  // const { name, description, page, size } = query;
+
+  // const search = {};
+  // if (name) search.name = name;
+  // if (description) search.description = description;
+
+  // const filter = ['name', 'description'];
+
+  const categories = await categoryService.getAll(request) || [];
+  return successF('Ok', categories, httpStatus.OK, response);
```
Comment (antobrines) [inline src/controllers/category.controller.js:45]: 'Ok' to 'OK'

File: src/controllers/role.controller.js [hunk 2]
```diff
@@ -1,26 +1,49 @@
...
+
+const update = async (request, response) => {
+  const { body, params } = request;
+  const { role_id } = params;
+  const role = await roleService.update(role_id, body);
+  if (!role) {
+    const error = new Error('Role not found');
+    return errorF(error, httpStatus.NOT_FOUND, response);
+  }
+  return successF('Ok', role, httpStatus.OK, response);
```
Comment (antobrines) [inline src/controllers/role.controller.js:19]: 'Ok' to 'OK'

File: src/controllers/role.controller.js [hunk 3]
```diff
@@ -1,26 +1,49 @@
...
+
+const getAll = async (request, response) => {
+  const { query } = request;
+  const { name, page, size } = query;
+
+  const search = {};
+  if (name) search.name = name;
+  const filter = ['name'];
+  const roles = await roleService.getAll(search, filter, page, size) || [];
+  return successF('Ok', roles, httpStatus.OK, response);
```
Comment (antobrines) [inline src/controllers/role.controller.js:41]: 'Ok' to 'OK'

File: src/controllers/event.controller.js [hunk 4]
```diff
@@ -1,173 +1,194 @@
...
+    response
+  );
+};
+
+const cancelEvent = async (request, response) => {
+  const { params } = request;
+  const { event_id } = params;
+  const event = await eventService.cancelEvent(event_id);
+  if (event.date_time.getFullYear() != 0) {
+    var error = new Error('Error in event anulation');
```
Comment (antobrines) [inline src/controllers/event.controller.js:138]: var to const

File: src/controllers/event.controller.js [hunk 5]
```diff
@@ -1,173 +1,194 @@
...
+  for (let i = 0; i < arrayEvent.length; i++) {
+    const category = await categoryService.FindOneById(arrayEvent[i].category);
+    arrayEvent[i]._doc.categoryComplet = category;
+
+    // remove all fileds don't need
+    delete arrayEvent[i]._doc.category;
+  }
+
+  if (arrayEvent.length == 0) {
+    var error = new Error('Event not found');
```
Comment (antobrines) [inline src/controllers/event.controller.js:161]: var to const

File: src/middlewares/digitalocean.middleware.js [hunk 6]
```diff
@@ -1,95 +1,91 @@
...
+
+    let params = {
+      Bucket: `${config.Aws.bucket}/${config.Aws.folder}/${sub_folder}`,
+      Key: namefile,
+      Body: file.data,
+      ACL: 'public-read',
+      ContentType: 'image/jpeg',
+    };
+
+    var putObjectPromise = s3_config.putObject(params).promise();
```
Comment (antobrines) [inline src/middlewares/digitalocean.middleware.js:62]: var to const

File: src/middlewares/filter.middleware.js [hunk 7]
```diff
@@ -1,55 +1,55 @@
...
+const errorF = require('../utils/error');
+const httpStatus = require('http-status');
+const jwt = require('jsonwebtoken');
+const config = require('../config');
+
+const filterF = (allowed = []) => (request, response, next) => {
+  const {
+    filter
+  } = request.query;
+  var isRight = true;
```
Comment (antobrines) [inline src/middlewares/filter.middleware.js:10]: var to let

File: src/middlewares/filter.middleware.js [hunk 8]
```diff
@@ -1,55 +1,55 @@
...
+  var filterTrue = {};
```
Comment (antobrines) [inline src/middlewares/filter.middleware.js:11]: var to let

File: src/services/event.service.js [hunk 9]
```diff
@@ -1,144 +1,140 @@
...
+const getAllFiltered = async (filter = {}) => {
+  return await Event.find(filter);
+};
+
+const findOneById = async (event_id) => {
+  return await Event.findById(event_id);
+};
+
+const submitParticipant = async (user_id, event_id) => {
+  var userParticipate = { user_id: user_id, status: constantes.STATUS_EVENT.VALIDATE };
```
Comment (antobrines) [inline src/services/event.service.js:36]: var to const

File: src/services/event.service.js [hunk 10]
```diff
@@ -1,144 +1,140 @@
...
+  return await Event.updateOne(
+    { _id: event_id },
+    { $push: { users: userParticipate } });
+};
+
+const cancelParticipant = async (user_id, event_id) => {
+  var userParticipate = { user_id: user_id };
```
Comment (antobrines) [inline src/services/event.service.js:43]: var to const

File: src/services/user.service.js [hunk 11]
```diff
@@ -1,109 +1,87 @@
...
+    userId: user._id
+  }, config.token.secret, { expiresIn: config.token.expire });
+
+  const test = await compareAsync(password, user.password);
+  if (!test) return { token: '', username: user.username };
+  return { token: accessToken, username: user.username };
+};
+
+const getAge = (birth_date) => {
+  var now = new Date();
```
Comment (antobrines) [inline src/services/user.service.js:68]: var to const for all var here

File: src/utils/Constantes.js [hunk 12]
```diff
@@ -1,48 +1,48 @@
...
-  STATUS_EVENT : {
-    VALIDATE : 'Validé',
-    PENDING : 'En attente de validation',
-    REFUSE : 'Refusé',
-  },
-  uuid,
-});
+const uuid = () => {
+  return 'xxxxxxxx-4xxx-yxxx'.replace(/[xy]/g, function (c) {
+    var r = (Math.random() * 16) | 0,
```
Comment (antobrines) [inline src/utils/Constantes.js:3]: var to const + replace Math.random() vulnaribilities

File: src/controllers/category.controller.js (hunk 1 above)
Comment (vysahjk) [reply src/controllers/category.controller.js]: fix in commit 032b41cc5ca74b8

File: src/controllers/role.controller.js (hunk 2 above)
Comment (vysahjk) [reply src/controllers/role.controller.js]: fix in commit 032b41cc5ca74b8

File: src/controllers/role.controller.js (hunk 3 above)
Comment (vysahjk) [reply src/controllers/role.controller.js]: fix in commit 032b41cc5ca74b8

File: src/controllers/event.controller.js (hunk 4 above)
Comment (vysahjk) [reply src/controllers/event.controller.js]: fix in commit 032b41cc5ca74b8

File: src/controllers/event.controller.js (hunk 5 above)
Comment (vysahjk) [reply src/controllers/event.controller.js]: fix in commit 032b41cc5ca74b8

File: src/middlewares/filter.middleware.js (hunk 8 above)
Comment (vysahjk) [reply src/middlewares/filter.middleware.js]: fix in commit 032b41cc5ca74b8

File: src/services/event.service.js (hunk 9 above)
Comment (vysahjk) [reply src/services/event.service.js]: fix in commit 032b41cc5ca74b8

File: src/services/event.service.js (hunk 10 above)
Comment (vysahjk) [reply src/services/event.service.js]: fix in commit 032b41cc5ca74b8

File: src/services/user.service.js (hunk 11 above)
Comment (vysahjk) [reply src/services/user.service.js]: fix in commit 032b41cc5ca74b8

File: src/utils/Constantes.js (hunk 12 above)
Comment (vysahjk) [reply src/utils/Constantes.js]: fix in commit 032b41cc5ca74b8

File: src/middlewares/digitalocean.middleware.js (hunk 6 above)
Comment (vysahjk) [reply src/middlewares/digitalocean.middleware.js]: fix in commit 032b41cc5ca74

File: src/middlewares/filter.middleware.js (hunk 7 above)
Comment (vysahjk) [reply src/middlewares/filter.middleware.js]: fix in commit 032b41cc5ca74b864aa0338a9cd57310b3a07db7

**Review by antobrines — APPROVED** [review]

**Pull request merged**